### DISCUSSION TITLE
`DriftAccessor` specification

### DISCUSSION THREAD

**Discussion by FaFre** [simolus3/drift#2931] [General]
I generate my database from a .drift file. Previously I was just using an `extension` on my database for dao's. Now I got aware of the DAO support and consider migrating my code. I don't really understand why it is necessary to specify `tables`, `queries` etc. in the annotation? In my case I have an indexer dao that basically fills all tables in the database from various different data sources. In a first test without specifying anything (= `@DriftAccessor()`) everything works as expected. I'm a bit worried that I might lead to inconsistencies later and I forget to update the annotation when working on the implementation. Do I have any downsides when just using the default annotation?

**Comment by simolus3** [comment]
If you're accessing tables through a central indexer or directly through the database, that's not a problem at all. `@DriftAccessor` is essentially a utility that generates getters for tables/views/drift files mentioned in the annotation. If you want to make use of that convenience, you can. If not, nothing bad is going to happen. You can even write your own `DatabaseAccessor` subclass without any `@DriftAccessor` annotation and it will work.

**Reply by FaFre** [reply]
Aha, now I get it! Thank you :)
